bluemira/base/constants.py

```diff
@@ -42,68 +64,218 @@
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    kelvin: Union[float, np.array, List[float]]
+        The temperature to convert [K]
+
+    Returns
+    -------
+    temp_in_celsius: Union[float, np.array]
+        The temperature [°C]
+    """
+    if np.any(np.less(kelvin, ABS_ZERO_K)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(kelvin, ureg.kelvin).to("celsius").magnitude
+
+
+def to_kelvin(celsius: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    temp_in_celsius: Union[float, np.array, List[float]]
+        The temperature to convert [°C]
+
+    Returns
+    -------
+    temp_in_kelvin: Union[float, np.array]
+        The temperature [K]
+    """
+    if np.any(np.less(celsius, ABS_ZERO_C)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(celsius, ureg.celsius).to("kelvin").magnitude
+
+
+def from_keV(
+    value: Union[float, np.array, List[float]],
+    to: str = "celsius",
+    *,
+    _from: str = "keV"
+):
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    value: Union[float, np.array, List[float]]
+        The temperature to convert [keV]
+    to: str
+        Unit to convert to eg celsius or kelvin
+    from: str
+        allows modification of keV prefix to eg eV or MeV etc
+
+    Returns
+    -------
+    temp: Union[float, np.array]
+        The temperature
+    """
+    return (ureg.Quantity(value, _from) / ureg.Quantity("k_B")).to(to).magnitude
+
+
+def kgm3_to_gcm3(density: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a density in kg/m3 to g/cm3
+
+    Parameters
+    ----------
+    density : Union[float, np.array, List[float]]
+        The density [kg/m3]
+
+    Returns
+    -------
+    density_gcm3 : Union[float, np.array]
+        The density [g/cm3]
+    """
+    return density / 1000.0
+
+
+def gcm3_to_kgm3(density: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a density in g/cm3 to kg/m3
+
+    Parameters
+    ----------
+    density : Union[float, np.array, List[float]]
+        The density [g/cm3]
+
+    Returns
+    -------
+    density_kgm3 : Union[float, np.array]
+        The density [kg/m3]
+    """
+    return density * 1000.0
+
+
+def pam3s_to_mols(flow_in_pam3_s):
+    """
+    Convert a flow in Pa.m^3/s to a flow in mols.
+
+    Parameters
+    ----------
+    flow_in_pam3_s: Union[float, np.array]
+        The flow in Pa.m^3/s to convert
+
+    Returns
+    -------
+    flow_in_mols: Union[float, np.array]
+        The flow in mol/s
+
+    Notes
+    -----
+    At 273.15 K for a diatomic gas
+    """
+    return flow_in_pam3_s / 2270
+
+
+def mols_to_pam3s(flow_in_mols):  # noqa :N802
+    """
+    Convert a flow in Pa.m^3/s to a flow in mols.
```

Comment:
```suggestion
    Convert a flow in mols to a flow in Pa.m^3/s.
```

Reply:
Shouldn't this say "mol/s" insteal of "mols"?

Reply:
Yep, I'll add it in #748